satpy/composites/__init__.py

```diff
@@ -1682,20 +1682,32 @@ class BackgroundCompositor(GenericCompositor):
     def __call__(self, projectables, *args, **kwargs):
         """Call the compositor."""
         projectables = self.match_data_arrays(projectables)
+
         # Get enhanced datasets
         foreground = enhance2dataset(projectables[0], convert_p=True)
         background = enhance2dataset(projectables[1], convert_p=True)
+        before_bg_mode = background.attrs["mode"]
+
         # Adjust bands so that they match
         # L/RGB -> RGB/RGB
         # LA/RGB -> RGBA/RGBA
         # RGB/RGBA -> RGBA/RGBA
```

Comment:
Are these cases still the same after your changes? Your tests suggest that `LA/RGB` will remain `RGB` instead of what is specified here (the second line) of `RGBA`.

Reply:
I'm not sure whether these comments are about the whole compositor or just `add_bands` it self...